site/docs/plugins/commands.md

```diff
@@ -5,10 +5,550 @@ next: false
+   } else next()
+})
+
+devCommands.command('usercount', 'Greetings', async (ctx, next) => {
+   if (ctx.from?.id === ctx.env.DEVELOPER_ID) {
+      await ctx.reply(
+        `Active users: ${/** Your business logic */}`
+    )
+   } else next()
+})
+
+devCommands.command('devlogout', 'Greetings', async (ctx, next) => {
+    if (ctx.from?.id === ctx.env.DEVELOPER_ID) {
+       await ctx.reply('Bye to me')
+       await ctx.setMyCommands(userCommands)
+    } else next()
+ })
+```
+
+```ts [users.ts]
+export const userCommands = new Commands<MyContext>();
+
+userCommands.command("start", "Greetings", async (ctx) => {
+  await ctx.reply("Hello user");
+  await ctx.setMyCommands(userCommands);
+});
+```
+
+```ts [types.d.ts]
+type MyContext = Context & CommandsFlavor<MyContext>;
+```
+
+:::
+
+Combining this knowledge with the following section will get your Command-game to the next level.
+
+## Scoped Commands
+
+Did you know you can allow different commands to be shown on different chats depending on the chat type, the language, and even the user status in a chat group?
+That's what Telegram call **Command Scopes**.
+
+The `Command` class returned by the `command` method exposes a method called `addToScope`.
+This method takes in a [BotCommandScope](/ref/types/botcommandscope) together with one or more handlers, and registers those handlers to be ran at that specific scope.
+
+You don't even need to worry about calling `filter`, the `addToScope` method will guarantee that your handler only gets called if the context is right.
+
+Here's an example of a scoped command:
+
+```js
+const myCommands = new Commands();
+
+myCommands
+  .command("start", "Initializes bot configuration")
+  .addToScope(
+    { type: "all_private_chats" },
+    (ctx) => ctx.reply(`Hello, ${ctx.chat.first_name}!`),
+  )
+  .addToScope(
+    { type: "all_group_chats" },
+    (ctx) => ctx.reply(`Hello, members of ${ctx.chat.title}!`),
+  );
+```
+
+The `start` command can now be called from both private and group chats, and it will give a different response depending on where it gets called from.
+Now if you call `myCommands.setCommands`, the `start` command will be registered to both private and group chats.
+
+Heres an example of a command that it's only accesible to group admins
+
+```js
+adminCommands
+  .command("secret", "Admin only")
+  .addToScope({
+    type: "all_chat_administrators",
+  }, async (ctx) => {
+    await ctx.reply("Free cake!");
+  });
+```
+
+Note: if you only want a command to be accesible on certain scopes, make sure you do not add a handler in the first `MyCommands.command` call. Doing that will automatically add it to all private chats, including groups.
+
+Here is an example of a command that it's only accesible in groups
+
+```js
+myCommands
+  .command(
+    "fun",
+    "Laugh",
+    /** skip this handler */
+  ).addToScope({
+    type: "all_group_chats",
+  }, async (ctx) => {
+    await ctx.reply("Haha");
+  });
+```
+
+## Command Translations
```

Comment:
how to use that feature with `@grammyjs/i18n`? We don't have access to the context object to pull the from .ftl file, is there any workaround to have all translations in one place?

Reply:
> how to use that feature with `@grammyjs/i18n`?

The `.localize` method only affects the command name and description. You should be able to use the i18n plugin as usual.
Example
```ts
myCommands.command("start", "start the bot", async (ctx) => {
  await ctx.reply(ctx.t("start-msg"));
}).localize("es","iniciar",'inicia el bot')
```

Reply:
Yeah, I wish to use i18n for the command and for the command description as well. 
I was looking at this example:
[setcommands.ts](https://github.com/bot-base/telegram-bot-template/blob/4d83d5969dc8c8185a31a1c7d0a956b3785cd320/src/bot/handlers/commands/setcommands.ts#L14-L21)
I thought `i18n` is accessible only from the bot context, but it was just exported.
Seems like it can be used something like in this example
```ts
function addLocalizations(command: Command, commandKey: string) {
  i18n.locales.forEach((locale) => {
    command.localize(
      locale,
      i18n.t(locale, `${commandKey}.command`),
      i18n.t(locale, `${commandKey}.description`)
    );
  });
  return command;
}

```

Reply:
I haven't used the i18n plugin, but this looks like it should work. I'd guess the ctx property just serves as a shortcut so you don't need to specify the location manually

Reply:
Nice, look's like I didn't understood your question quite correctly. 
Looking at your example, might be handy to have a getter for the `_commands` private field (which is an array of all the Command classes contained in a Commands instance), so you can pass easily each command into a function like this.

Reply:
hmm the .command method returns the command instance tho, so I don't know if this is really necessary
wouldn't hurt tho

Reply:
> the .command method returns the command instance

Yeah, but might also be handy for stuff like: (following the above example)
```ts
// register
const all = myCommands.commands
all.forEach( (command ) => addLocalizations(command, command.name))
```

Reply:
@carafelix mind turning this suggestion into an issue too, please?

Reply:
It's already addressed in the `addEntities` branch?